jetty-websocket/websocket-server/src/main/java/org/eclipse/jetty/websocket/server/WebSocketServerFactory.java

```diff
@@ -562,6 +561,12 @@ private boolean upgrade(HttpConnection http, ServletUpgradeRequest request, Serv
         {
             throw new IllegalStateException("Not a 'WebSocket: Upgrade' request");
         }
+
+        if ("HTTP/2".equals(request.getHttpVersion()))
+        {
+            LOG.warn("WebSocket Bootstrap from HTTP/2 (RFC8441) not supported in Jetty 9.x");
+        }
+
```

Comment:
This additional code is not needed, since if it's a HTTP/2 request it will enter the `if` statement below and throw.

Reply:
See issue #3530

Reply:
@joakime I'm aware of that issue. I don't understand why an `IllegalStateException` thrown with a proper message is swallowed - that should be plenty enough warnings to users.

I would rather write a test case, because it may be that we never reach this point of the code because we fail or take other paths before arriving here.